### PR TITLE
Feature: add donation confirmation settings UI

### PR DESCRIPTION
**Pull request by jonwaldstein**
## Description

This updates the donation confirmation settings UI with a popover and list of available template tags (pulled from v2 forms).

## Affects

The donation confirmation settings UI in the form builder.  Also the confirmation page now uses Interweave to display the rich text from the template tags that were generated.

## Visuals

Screenshot 2023-08-03 at 12 40 15 PM

## Testing Instructions

- Build assets and view a donation form builder
- Interact with the donation confirmation settings
- Make sure v2 form template tags work as expected

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [ ] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1203889335107900

### CODE REVIEW

**Comment by jonwaldstein** [comment]
@glaubersilva this ready for review.

**Review by glaubersilva — CHANGES_REQUESTED** [review]
@jonwaldstein Great work! About the code, I left just one comment about a minor change. 

But about the feature itself, we have a few things to take care of:

1. The line breaks don't work on the confirmation page
2. The P2P tags are loaded, but next gen will be released without compatibility with the v3 forms, more details here: https://github.com/impress-org/give-peer-to-peer/pull/505
3. The PDF receipt link is using the same color as the header background and because of that seems to not be present at the first look. 

Check the attached screenshot for visual reference:

tag-problems

**Comment by jonwaldstein** [comment]
@glaubersilva thanks for brining these up.  I discovered a few things:

1. _The line breaks don't work on the confirmation page_: they actually do! but some tags like pdf_receipt is making it not work.  It might be a setting on Interweave or the way we handle converting tags with links. I'll look into that.
2. _The P2P tags are loaded, but next gen will be released without compatibility with the v3 form..._ we can try to filter those out - i'll look into that.
3. Ah yeah, we can add a little styling help for links on the classic design

File: src/FormBuilder/resources/js/form-builder/src/settings/donation-confirmation/index.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    } = usePopoverState();
+
+    const {
+        isOpen: isDescriptionMessageSettingsOpen,
+        toggle: toggleDescriptionMessageSettings,
+        close: closeDescriptionMessageSettings,
+    } = usePopoverState();
+
+    return (
+        <PanelBody title={__('Donation Confirmation')} initialOpen={false}>
```
[the remark this answers is not in the document]
Comment (jonwaldstein) [reply src/FormBuilder/resources/js/form-builder/src/settings/donation-confirmation/index.tsx]: resolved 4bab9e37

**Comment by jonwaldstein** [comment]
@glaubersilva okay man I fixed those main issues.

- Added nl2br for the content so interweave can handle line breaks appropriately alongside anchor tags
- Filtered available tag list
- Styling for me was okay on the link, let's add a QA item to make the confirmation page smarter for contrast in background color and text.  Right now it's going to depend on the design options.

Screenshot 2023-08-09 at 4 57 04 PM

**Comment by glaubersilva** [comment]
@jonwaldstein Thanks for the changes, now the line breaks are working, but the link keeps the same background color on my end. About the P2P tags, a weird bug is happening when the addon is activated, check the video to see what's happening: https://www.loom.com/share/ec528d56ae124311809fcd6067b80ffe

**Comment by jonwaldstein** [comment]
> @jonwaldstein Thanks for the changes, now the line breaks are working, but the link keeps the same background color on my end. About the P2P tags, a weird bug is happening when the addon is activated, check the video to see what's happening: https://www.loom.com/share/ec528d56ae124311809fcd6067b80ffe

@glaubersilva I just pushed a fix for the p2p conflict.  please try again.

**Pull request merged**